.github/workflows/auto-spotless-check.yml

```diff
@@ -0,0 +1,53 @@
+name: Auto spotless check
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Set up JDK for running Gradle
+        uses: actions/setup-java@c5195efecf7bdfc987ee8bae7a71cb8b11521c00 # v4.7.1
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Set up gradle
+        uses: gradle/actions/setup-gradle@06832c7b30a0129d7fb559bcc6e43d26f6374244 # v4.3.1
+        with:
+          cache-read-only: true
+
+      - name: Check out PR branch
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: gh pr checkout ${{ github.event.pull_request.number }}
+
+      - name: Spotless
+        run: ./gradlew spotlessApply
```

Comment:
nit: the workflow called `spotless-check` is the one that runs `spotlessApply` and the workflow called `spotless-apply` doesn't. It's only slightly confusing, not a huge deal.

Reply:
yeah, it's named more for what you see in the github U/X, I'll probably make it all more generic and rename things again later

Spotless Check runs on PRs, generates the patch file, then Spotless Apply runs afterwards and applies the patch file